src/main/java/sg/ncl/DataController.java

```diff
@@ -0,0 +1,237 @@
+        dataObject.put("accessibility", dataset.getAccessibility());
+        dataObject.put("resources", new ArrayList());
+        dataObject.put("approvedUsers", new ArrayList());
+        dataObject.put("releasedDate", dataset.getReleasedDate());
+        log.debug("DataObject: {}", dataObject.toString());
+
+        HttpEntity<String> request = createHttpEntityWithBody(dataObject.toString());
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+
+        ResponseEntity response = getResponseEntity(id, request);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                switch (exceptionState) {
+                    case DATASET_NAME_IN_USE_EXCEPTION:
+                        log.warn("Dataset name already exists.");
+                        model.addAttribute(MESSAGE_ATTRIBUTE, "Error(s):<ul><li>dataset name already exists</li></ul>");
+                        break;
+                    case FORBIDDEN_EXCEPTION:
+                        log.warn("Saving of dataset forbidden.");
+                        model.addAttribute(MESSAGE_ATTRIBUTE, "Error(s):<ul><li>saving dataset forbidden</li></ul>");
+                        break;
+                    default:
+                        log.warn("Unknown error.");
+                }
+
+                return CONTRIBUTE_DATA_PAGE;
+            }
+        } catch (IOException e) {
+            log.error("validateContributeData: {}", e.toString());
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return "redirect:/data";
+    }
+
+    @RequestMapping("/remove/{id}")
+    public String removeDataset(@PathVariable String id, RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getDataset(id), HttpMethod.DELETE, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                if (exceptionState == FORBIDDEN_EXCEPTION) {
+                    log.error("Removing of dataset forbidden.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                }
+            }
+        } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)